webapp/components/utils/Notifications.js

```diff
@@ -40,4 +40,117 @@ export const notifications = [
       },
     },
   },
+  {
+    read: false,
+    reason: 'filed_report_on_resource',
+    from: {
+      __typename: 'Report',
+      id: 'reportOnUser',
+      filed: [
+        {
+          reasonCategory: 'discrimination_etc',
+          reasonDescription: 'This user is harassing me with bigoted remarks!',
+          reportedResource: {
+            __typename: 'User',
+            id: 'badWomen',
+            slug: 'mrs.-badwomen',
+            name: 'Mrs. Badwomen',
+          },
+        },
+      ],
+    },
+  },
 ]
+
+export const extractNotificationDataOfCurrentUser = (notification, currentUser) => {
+  const from = notification.from
+  let triggerer
+  const id = from.id
+  const createdAt = notification.createdAt
+  const read = notification.read
+  const reason = notification.reason
+  let title
+  let author
+  let user = null
+  let post = null
+  let comment = null
+  let contentExcerpt = null
+  let report = null
+  let reasonExtention = ''
+
+  if (from.__typename === 'Post') {
+    post = from
+    triggerer = post.author
+  } else if (from.__typename === 'Comment') {
+    comment = from
+    triggerer = comment.author
+    post = comment.post
+  } else if (from.__typename === 'Report') {
+    report = {
+      reasonCategory: from.filed[0].reasonCategory,
+      reasonDescription: from.filed[0].reasonDescription,
+    }
+    triggerer = currentUser
+    if (from.filed[0].reportedResource.__typename === 'User') {
+      user = from.filed[0].reportedResource
+      reasonExtention = '.user'
+    } else if (from.filed[0].reportedResource.__typename === 'Post') {
+      post = from.filed[0].reportedResource
+      reasonExtention = '.post'
+    } else if (from.filed[0].reportedResource.__typename === 'Comment') {
+      comment = from.filed[0].reportedResource
+      post = from.filed[0].reportedResource.post
+      reasonExtention = '.comment'
+    }
+  }
+
+  if (user) {
+    title = user.name
+    author = user
+  } else {
+    title = post.title
+    if (comment) {
+      author = comment.author
+      contentExcerpt = comment.contentExcerpt
+    } else {
+      author = post.author
+      contentExcerpt = post.contentExcerpt
+    }
+  }
+
+  const params = user
+    ? {
+        id: user.id,
+        slug: user.slug,
+      }
+    : post
+    ? {
+        id: post.id,
+        slug: post.slug,
+      }
+    : {}
+  const hashParam = comment ? { hash: `#commentId-${comment.id}` } : {}
+  const linkTo = {
+    name: user ? 'profile-id-slug' : 'post-id-slug',
+    params,
+    ...hashParam,
+  }
+
+  const data = {
+    triggerer,
+    id,
+    createdAt,
+    read,
+    reason,
+    title,
+    user,
+    post,
+    comment,
+    contentExcerpt,
+    author,
+    report,
+    reasonExtention,
+    linkTo,
+  }
+  return data
+}
```

Comment:
wow! this is complicated! I gotta believe most of this is the backends job, oder?
I'd have to play around with it/wait for other people's opinions, but seems like the frontend is doing a lot of extra work here :worried:

Reply:
Hm is it so complicated?
A lot of this code was in the Vue components directly before. And because the hierarchy gets deeper with the `Report`notification …
… it extracts the data into a flat hierarchy.
This way the Vue components can be much less complex.

If you put it some way in the backend then it will look the same and you have kind of logic there that is only related to the way we display the data in the frontend.

I could write some comments to give a little hint of the ideas of this extraction? @mattwr18 

I’m curious to other ones opinions @Human-Connection/hc-dev-team ?

Reply:
Yes! Too complicated. I didn't understand enough to tell if this is the job of the backend or the frontend though.

Reply:
I would like to discuss this personal with you. @mattwr18